doc/source/serve/production-guide/kubernetes.md

```diff
@@ -0,0 +1,182 @@
+
+The first step is to install the `KubeRay` operator into your Kubernetes cluster.
+This creates a pod that runs the `KubeRay` controller. The `KubeRay` controller manages resources based on the `RayService` CRs you create.
+
+Install the operator using `kubectl apply` and check that the controller pod is running:
+```console
+$ kubectl create -k "github.com/ray-project/kuberay/manifests/cluster-scope-resources?ref=v0.3.0"
+$ kubectl apply -k "github.com/ray-project/kuberay/manifests/base?ref=v0.3.0"
+$ kubectl get deployments -n ray-system
+NAME           READY   UP-TO-DATE   AVAILABLE   AGE
+ray-operator   1/1     1            1           40s
+
+$ kubectl get pods -n ray-system
+NAME                            READY   STATUS    RESTARTS   AGE
+ray-operator-75dbbf8587-5lrvn   1/1     Running   0          31s
+```
+
+For more details, see the [KubeRay quickstart guide](kuberay-quickstart).
+
+## Deploying a Serve application
+
+Once the KubeRay controller is running, you can manage your Ray Serve application by creating and updating a `RayService` custom resource (CR).
+`RayService` custom resources consist of the following:
+- a `KubeRay` `RayCluster` config defining the cluster that the Serve application runs on.
+- a Ray Serve [config](serve-in-production-config-file) defining the Serve application to run on the cluster.
+
+When the `RayService` is created, the `KubeRay` controller first creates a Ray cluster using the provided configuration.
+Then, once the cluster is running, it deploys the Serve application to the cluster using the [REST API](serve-in-production-deploying).
+The controller also creates a Kubernetes Service that can be used to route traffic to the Serve application.
+
+Let's see this in action by deploying the same [`FruitStand` example](serve-in-production-example) from above.
+The Serve config for the example is embedded into [this example `RayService` CR](https://github.com/ray-project/kuberay/blob/release-0.3/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml).
+To follow along, save this CR locally in a file named `ray_v1alpha1_rayservice.yaml`:
+
+```console
+curl -o ray_v1alpha1_rayservice.yaml https://raw.githubusercontent.com/ray-project/kuberay/release-0.3/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml
+```
+
+To deploy the example, we simply `kubectl apply` the CR.
+This creates the underlying Ray cluster, consisting of a head and worker node pod (see {ref}`_cluster-key-concepts-under-construction` for more details on Ray clusters), as well as the service that can be used to query our application:
+
+```console
+$ kubectl apply -f ray_v1alpha1_rayservice.yaml
+
+$ kubectl get rayservices
+NAME                AGE
+rayservice-sample   7s
+
+$ kubectl get pods
+NAME                                                      READY   STATUS    RESTARTS   AGE
+ervice-sample-raycluster-qd2vl-worker-small-group-bxpp6   1/1     Running   0          24m
```

Comment:
> ervice-sample-raycluster-qd2vl-worker-small-group-bxpp6   1/1     Running   0

I think something got cut off at the start of this line.

Reply:
ah yeah k8s does that

Reply:
fixed